drivers/espi/espi_emul.c

```diff
@@ -87,6 +91,44 @@ static bool espi_emul_get_channel_status(const struct device *dev, enum espi_cha
 	return (data->cfg.channel_caps & ch);
 }
 
+#ifdef CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION
+static int espi_emul_set_acpi_shm(const struct device *dev, uint8_t *data,
+				  size_t size)
+{
+	struct espi_emul_data *dev_data = dev->data;
+
+	if (size > CONFIG_ESPI_EMUL_HOST_ACPI_SHM_REGION_SIZE)
+		return -EIO;
+	memcpy(dev_data->shm_acpi_mmap, data, size);
+	return 0;
+}
+#endif
+
+static int espi_emul_read_lpc_request(const struct device *dev,
+				      enum lpc_peripheral_opcode op,
+				      uint32_t *data)
+{
+	struct espi_emul_data *dev_data = dev->data;
+
+	switch (op) {
+#ifdef CONFIG_ESPI_PERIPHERAL_ACPI_SHM_REGION
+	case EACPI_GET_SHARED_MEMORY:
+		*data = (uint32_t)dev_data->shm_acpi_mmap;
```

Comment:
Here, we should send a request for the pointer of the mmap to the host via lpc_reqest, as it is done in espi_emul_receive_vwire.

Reply:
Done.